File: materials-evaluation-dotnet/Database/Models/Batch.cs
using MaterialsEvaluation.Modules.QualityEvaluation.Domain;

namespace MaterialsEvaluation.Database
{
    public class Batch
    {
        public Guid Id { get; set; }

        public Guid MaterialId { get; set; }
        public virtual Material Material { get; set; }

        public Guid QualityVisionId { get; set; }
        public virtual QualityVision QualityVision { get; set; }

        public DateTime CreatedAt { get; set; }
        public int AmountOfTests { get; set; }
        public DateTime? CalculatedAt { get; set; }
        public string Status { get; set; }

        public List<Test> Tests { get; set; }

        public Batch(
            Guid id,
            Guid materialId,
            Guid qualityVisionId,
            DateTime createdAt,
            int amountOfTests,
            DateTime? calculatedAt,
            string status
        )
        {
            Id = id;
            MaterialId = materialId;
            QualityVisionId = qualityVisionId;
            CreatedAt = createdAt;
            AmountOfTests = amountOfTests;
            CalculatedAt = calculatedAt;
            Status = status;
        }
    }
}

File: materials-evaluation-spa/src/app/models/Batch.ts
import { Material } from './Material';
import { QualityProperty } from './QualityProperty';
import { QualityVision } from './QualityVision';

export enum Status {
  Pending,
  OutOfRange,
  InRange,
}

export interface Test {
  qualityProperty: QualityProperty;
  resultQualitative: boolean | null;
  resultQuantitative: number | null;
}

export interface Batch {
  id: string;
  material: Material;
  qualityVision: QualityVision;
  createdAt: string;
  amountOfTests: number;
  calculatedAt: string | null;
  status: Status;
  tests: Test[];
}
